can/interfaces/vector/canlib.py

```diff
@@ -154,31 +188,64 @@ def set_filters(self, can_filters=None):
 
     def recv(self, timeout=None):
         end_time = time.time() + timeout if timeout is not None else None
-        event = vxlapi.XLevent(0)
-        event_count = ctypes.c_uint()
+
+        if self.fd:
+            event = vxlapi.XLcanRxEvent()
+        else:
+            event = vxlapi.XLevent()
+            event_count = ctypes.c_uint()
+            
         while True:
-            event_count.value = 1
-            try:
-                vxlapi.xlReceive(self.port_handle, event_count, event)
-            except VectorError as exc:
-                if exc.error_code != vxlapi.XL_ERR_QUEUE_IS_EMPTY:
-                    raise
+            if self.fd:
+                try:
+                    vxlapi.xlCanReceive(self.port_handle, event)
+                except VectorError as exc:
+                    if exc.error_code != vxlapi.XL_ERR_QUEUE_IS_EMPTY:
+                        raise
+                else:
+                    if event.tag == vxlapi.XL_CAN_EV_TAG_RX_OK or event.tag == vxlapi.XL_CAN_EV_TAG_TX_OK:
+                        msg_id = event.tagData.canRxOkMsg.canId
+                        dlc = self.map_dlc(event.tagData.canRxOkMsg.dlc)
+                        flags = event.tagData.canRxOkMsg.msgFlags
+                        timestamp = event.timeStamp * 1e-9
+                        msg = Message(
+                            timestamp=timestamp + self._time_offset,
+                            arbitration_id=msg_id & 0x1FFFFFFF,
+                            extended_id=bool(msg_id & vxlapi.XL_CAN_EXT_MSG_ID),
+                            is_remote_frame=bool(flags & vxlapi.XL_CAN_RXMSG_FLAG_RTR),
+                            is_error_frame=bool(flags & vxlapi.XL_CAN_RXMSG_FLAG_EF),
+                            is_fd=True,
```

Comment:
As I understood it, this should be `bool(flags & vxlapi.XL_CAN_RXMSG_FLAG_EDL)` since EDL is set for all FD-frames to distinguish them from regular frames. Or what do you think?

Reply:
I guess I messed up fd for channel and fd per message. You're right!